prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -341,6 +365,28 @@ fun Bundle.filterObservations(
     return filteredBundle
 }
 
+/**
+ * Filter out observations that pass the condition filter for a [receiver]
+ * The [bundle] and [shortHandLookupTable] will be used to evaluate whether
+ * the observation passes the filter
+ *
+ * @return copy of the bundle with filtered observations removed
+ */
+fun Bundle.filterMappedObservations(
+    conditionFilter: ReportStreamConditionFilter,
+): Bundle {
+    val codes = conditionFilter.codes()
+    val observations = this.getObservations()
+    val toKeep = observations.filter { it.getMappedConditions().any(codes::contains) }.map { it.idBase }
+    val filteredBundle = this.copy()
+    observations.forEach {
+        if (it.idBase !in toKeep) {
+            filteredBundle.deleteResource(it)
```

Comment:
I agree in, but there wasn't any existing logging for pruned resources (in the legacy `conditionFilter`). Any thoughts @arnejduranovic? Should we log when individual resources are pruned? I don't think it will add much value and it will introduce a lot of logging per receiver.

Right now we only log if the entire bundle is not sent due to the `conditionFilter` or `mappedConditionFilter`

Whatever we decide, we should make the change in the upcoming PR that combines the two filters to avoid writing that logic three times.